spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriptionProvider.java

```diff
@@ -57,7 +60,7 @@
 	}
 
 	SpanSubscription<T> newCoreSubscriber(Tracing tracing) {
-		return new SpanSubscriber<>(this.subscriber, this.context, tracing, this.name);
+		return new SpanSubscriber<>(this.subscriber, this.context, tracing, this.scannable.name());
```

Comment:
`SpanSubscriber` could also benefit from an on-demand `scannable.name()`

Reply:
SpanSubscriber uses `name`  in the constructor, so I think passing `scannable` to `SpanSubscriber` is meaningless.

Reply:
yes, you're right! 👍